### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by Victoria129**
- I have added the following features to the application.
- [ ] Routing.
- [ ] Filtering.
- [ ] Styling.

### CODE REVIEW

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @Victoria129 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## ❌ Invalid Code Review Request

You have submitted a project, that that does not have the Miroverse-provided readme template. Kindly add this[ readme](https://github.com/microverseinc/readme-template), and customize it to your project by adding relevant information about your project in all the readme sections. Please do not delete any section (the FAQ section is optional).

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @Victoria129, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't use the recommended [readme](https://github.com/microverseinc/readme-template) template. The project also lacks litter checks. Please make the necessary changes.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

### Martin Afani
#### Slack: whiz25

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @Victoria129 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- Good commit history ✔️
- Good PR description ✔️
- Your README file is professional ✔️

## Required Changes ♻️
_Check the comments under the review._

- you should record a video presenting the features of the project.
- The maximum length of the video is 5 minutes.
- The video should contain a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.
- The video should contain a demo of the project, highlighting all the project features.
- The video should contain a highlight about a piece of the code the student finds interesting or is proud of making.
- you should deploy the application using Heroku, Netlify, or GitHub Pages.
-  The project should be deployed properly, and the home and detail page works as expected.
-  A link to the deployed project is present in the README.
- The README should contain an "Acknowledgements" section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).The README contains an "Acknowledgements" section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 1]
```diff
@@ -1 +1,227 @@
...
+
+After you're finished please remove all the comments and instructions!
+-->
+
+<div align="center">
+
+  <img src="murple_logo.png" alt="logo" width="140"  height="auto" />
+  <br/>
+
+</div>
```
Comment (amejid) [inline README.md]: - [ ] Please remove these lines as they are rendering a broken image.

Screenshot from 2023-03-17 17-11-14

File: README.md [hunk 2]
```diff
@@ -1 +1,227 @@
...
+```
+  git clone https://github.com/Victoria129/Air-Pollution.git
+```
+
+### Prerequisites
+
+In order to run this project you need:
+
+``` node
+```
```
Comment (amejid) [inline README.md]: - [ ] These lines show an empty section.

Screenshot from 2023-03-17 17-10-39

File: README.md [hunk 3]
```diff
@@ -1 +1,227 @@
...
+
+
+### Deployment
+
+You can deploy this project using:
+
+
+```npm run dev
+
+```
```
Comment (amejid) [inline README.md:156]: Same thing here

Screenshot from 2023-03-17 17-10-51

File: README.md [hunk 4]
```diff
@@ -1 +1,227 @@
...
+## ⭐️ Show your support <a name="support"></a>
+
+If you like this project please give it a star!
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
```
Comment (amejid) [inline README.md:201]: The Acknowledgment section is empty, I think you should mention the author of the original design here.

Screenshot from 2023-03-17 17-09-26

**Review by amejid — COMMENTED** [review]
Hi @Victoria129, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have a professional README and linters are not running. There are sections that are empty. Kindly fill them with relevant information.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by GedeonTS — CHANGES_REQUESTED** [review]
# INVALID REVIEW REQUEST

Hi @Victoria129 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have linter checks.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/pages/home-page.jsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    {
+
+   countriesArray.map((country)=>{
+    // console.log(country)
+  return  <CityCard   key={country.name} name={country.name} continent={country.continent} flag={country.flag} lat={country.lat} lon={country.lon}/>
+})
+    }
+  </div>
+  </>
+)
```
Comment (Abenezer-Tilahun) [inline src/pages/home-page.jsx]: - [ ] As stated in the project requirement in the interaction part for the Home page it state that **Along with the category name, you will display some numeric values per category that come from the API.** So Kindly try to implemented it .
Capture

- Your Project Home Page design 

Capture

- The design guidelines Home page 

Capture

File: src/pages/home-page.jsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+export default HomePage;
```
Comment (Abenezer-Tilahun) [inline src/pages/home-page.jsx]: - [ ] As stated in the project requirement You should follow these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), including: Colors (select one main color for your website), Typography: font face, size and weight, Layout: composition and space between elements. But I notice that your design is a little bit different from the design guidelines Kindly i recommend you to follow the design guidelines for the Homepage and adjust  **Colors (select one main color for your website**) you used 2 color for the body(rgb(221, 211, 211)) and the container(rgb(240, 91, 116)) and **Layout: composition and space between elements.** Just like the design guidelines 
- For the Home page the design guidelines has the following Layout: composition and space between elements, the category are adjust in the 2 columns with different color composition 
Capture

- Your Project design Look likes this 

Capture

File: src/tests/cityCard.test.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+import React from 'react';
+import CityCard from '../components/CityCard';
+
+const fakeData = {
+  name: 'name', continent: 'continent', flag: 'flag', lat: 'lat', lon: 'lon',
+};
+
+test('should render CityCard', () => {
+  render(<CityCard props={fakeData} />);
+});
```
Comment (Abenezer-Tilahun) [inline src/tests/cityCard.test.js]: - [ ] Good job for Create integration tests for your application using the React Testing Library. But Kindly makes sure that your test cases are passing the requirement , So kindly fix it.

Capture

**Review by Abenezer-Tilahun — CHANGES_REQUESTED** [review]
Hi @Victoria129   , 

Great job for implemented  the some of the project requirement  , but there are some changes that need to be implemented in order to proceed to the next project 👍

## Highlights 👏🏻

✔️ No Linters error
✔️ You used correct Git-Flow.
✔️ Good commit messages
✔️ Professional READEME 


almost

## Required Changes ♻️
_Check the comments below the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/CityCard.jsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  const handleNavigation = (city) => {
+    // dispatch(fetchPollutionData({lat,lon}))
+    navigate(`${city}`,{state:{name,continent,lat,lon}})
+  }
+  return (
+    <div className='city-card' onClick={() => handleNavigation('city')}>
+    <h4>{name}</h4>
+    <p>lat : {lat}</p>
+    <img src={flag} alt="country-flag" />
+    </div>
```
Comment (codepantha) [inline src/components/CityCard.jsx]: - [ ] Please let's try to stick to the design guidelines for the cards as shown in the image below 👇 
- [ ] Please put the arrow at the top-right of the card
### DESIGN GUIDELINE
image

### YOUR DESIGN
image

File: src/pages/home-page.jsx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+  let dispatch = useDispatch()
+const countriesArray = useSelector(selectAllCountries)
+// console.log("All Countries", countriesArray)
+
+const [continent, setContinent] = useState("All")
+
+const handleContinentChange=(continent)=>{
+  // console.log("Button clicked")
+  // setContinent(continent)
```
Comment (codepantha) [inline src/pages/home-page.jsx]: - [ ] Please remove commented-out code as they make your codebase look cluttered 👍

File: src/components/Pollution.jsx [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  <p>no : {pollutionData?.no}</p>
+  <p>co : {pollutionData?.co}</p>
+  <p>no2 : {pollutionData?.no2}</p>
+  <br />
+  <p>Air Quality:  {pollutionData?.stringQuality}</p>
+</div>
+</div>
+
+<button onClick={handleGoBackButton}>Go Back</button>
+</>
```
Comment (codepantha) [inline src/components/Pollution.jsx]: - [ ] According to the design guideline, please place the "Go Back" button at the top in the form of a navigation
- [ ] In the navigation, also include a Microphone and Settings icon to be consistent with the design
### Design Guidelines
image
### Your Design
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @Victoria129 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/pages/home-page.jsx [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+  <div className="container">
+    {
+
+   countriesArray.map((country)=>{
+  return  <CityCard   key={country.name} name={country.name} continent={country.continent} flag={country.flag} lat={country.lat} lon={country.lon}/>
+})
+    }
+  </div>
+  </>
+)
```
Comment (omar-muhamad) [inline src/pages/home-page.jsx:41]: - [ ] As requested by the previous reviewers, Kindly stick to the given design and fix the overflowing city cards on the home page, and It's supposed to be two columns only not 3 as you did.

| Given Design  | Your Design |
| ------------- | ------------- |
| image  | image  |

[Hint] I suggest using `display: grid;` like this:
```css
display: grid;
grid-template-columns: repeat(2, 1fr);
```

File: src/components/Pollution.jsx [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+  <p>no : {pollutionData?.no}</p>
+  <p>co : {pollutionData?.co}</p>
+  <p>no2 : {pollutionData?.no2}</p>
+  <br />
+  <p>Air Quality:  {pollutionData?.stringQuality}</p>
+</div>
+</div>
+</>
+  )
+}
```
Comment (omar-muhamad) [inline src/components/Pollution.jsx]: - [ ] As requested by the previous reviewers, Kindly stick to the given design and make your details page take the full width of the page and fix all elements' placement, widths, and margins to match the given design.

| Given Design  | Your Design |
| ------------- | ------------- |
| image  | image  |

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Victoria129,

#### Nice job making the required changes. Unfortunately, you missed some of them but you are almost there.



## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Ismailco — APPROVED** [review]
### Hi @Victoria129, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

cr-approve

## Highlights

✔️ Make required changes
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

Good luck! 🎉 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Comment by Victoria129** [comment]
Thank you so much! @Ismailco

**Pull request merged**